### ISSUE TITLE
Feature request: Get option to color point by z-depth

### ISSUE THREAD

**Issue by jo-chemla** [CloudCompare/CloudCompare#505]
In the Properties dialog of a point cloud, the option Color could be added the Z-depth ability (after RGB and Scalar), being also able to tune the min and max values. The difference with computing a scalar field storing this depth is that it would adapt to the viewport - and could be exported as an animation for example.

**Comment by dgirardeau** [comment]
Do you mean to use the 'depth map' as produced by the graphic card?

**Comment by jo-chemla** [comment]
Yep, use it 'as is', as stored in the z buffer of the GPU. Being able to set min/max and corresponding color scale just as when using scalar field would be a plus. 
As a sidenote, I've discovered the "create sensor camera" function, it works great and makes the process of set up of the animation easier. A nice addition would be the ability to duplicate cameras, in order to set keyframe viewports separated by a given rotation/translation.

**Comment by jo-chemla** [comment]
Would it be possible to also give the ability to use a "Normal" shader, mapping points normals direction to RGB ? Giving results like this : https://i.stack.imgur.com/WZEsy.jpg

**Comment by rjanvier** [comment]
There is already a normal to HSV option but you need something more "screen space".

**Comment by jo-chemla** [comment]
Yes, I forgot to mention that, it should also be relative to the camera viewport - like the z-depth computed in screen space. Thanks for your point.

**Comment by rjanvier** [comment]
there are such shaders in Meshlab.

**Comment by rjanvier** [comment]
Render > shader > depthmap / Render > shaders > normalmap

**Comment by jo-chemla** [comment]
Yes but I cannot import a 500M point cloud/mesh in Meshlab - which Cloudcompare can easily handle. I am interested in applying this shader to a point cloud - but will accept any solution based on a mesh since my point cloud is simply made of the vertices of a ~250M triangles.

**Comment by rjanvier** [comment]
If you are "simply" trying to enhance your point cloud maybe you should use the PCV plugin. Else, I am afraid that CloudCompare does not include those shaders yet. It's an interesting enhancement but I think it's not on our top priority list for now. CloudCompare is an Open Source project so if you want to contribute fill free to open a pull request.

**Comment by dgirardeau** [comment]
Have you tried the brand new version of Meshlab? I heard it was much more powerful (especially for big meshes)

**Comment by jo-chemla** [comment]
Hi daniel, I've tried on a decent config (2xTitanX and 2 Xeon E5) and the model caan be imported into the new version of Meshlab, and indeed another shader like the depth can be used, but I cannot rotate/pan - the interface is becoming way too slow when the xyz is imported. Thanks for the help. If you point me at the right portion of the code (I will look for the code doing UI for the RGB/Scalar field to modify and check one shader example), I think I can make these changes and make a PR.

**Comment by rjanvier** [comment]
Have you tried to increase the max amount of memory dedicated to geometry:
in "Tools > options" then restart meshlab?

**Comment by dgirardeau** [comment]
I really wonder how one can do this 'properly' as the SF editor is a 'property' of a single cloud (see https://github.com/CloudCompare/CloudCompare/blob/master/qCC/db_tree/ccPropertiesTreeDelegate.cpp#L550), and shaders are currently attached to a 3D view (see https://github.com/CloudCompare/CloudCompare/blob/master/libs/qCC_glWindow/ccGLWindow.cpp#L2667) ...

The only solution I see would be to be able to attach a shader to a point cloud (or to any entity) and to let the shader provide its own property editor (widget). In your case it could be similar to the existing SF editor, with a few modifications however. And all this would be integrated in the property editor with a generic interface... However I doubt such a shader is worse the pain (and I won't accept quick and dirty code here ;).

Maybe a more reasonable option would be to create a new ccGLFilter plugin (like qEDL and qSSAO) that displays a dialog similar to the SF editor when the user clicks on its icon. At least it seems much more realistic to me.

**Comment by jo-chemla** [comment]
Yep I did that, with no luck. Thanks again.
